homeassistant/helpers/entity.py

```diff
@@ -178,6 +179,23 @@ class DeviceInfo(TypedDict, total=False):
     default_model: str
 
 
+@dataclass
+class EntityDescription:
+    """An class that describes Home Assistant entities."""
+
+    # This is the key identifier for this entity
+    key: str
```

Comment:
I'm not a huge fan of putting the `key` in the `EntityDescription` itself. For some integrations it might make sense to have the `key` in the description data. [nws](https://github.com/cdce8p/ha-core/blob/beccf65fa64695bab6adcdb4018f9a66bfd1c18e/homeassistant/components/nws/sensor.py#L43-L53) would be one of them.

However there are also some that actually use the key to get the descriptions for only those sensors they want to add based on the `monitored_conditions`. [rova](https://github.com/cdce8p/ha-core/blob/f0382d48b763ddf19f02cfff97e5e5cef6e2f74a/homeassistant/components/rova/sensor.py#L100-L102) would be an example for that.

I would like to suggest to either add it to the individual integration sensor description or make it optional.

Reply:
> I'm not a huge fan of putting the key in the EntityDescription itself. For some integrations it might make sense to have the key in the description data

It is a lot actually...

Required or not... dunno. I discussed this on chat with @balloob actually, I think it makes sense to have something required that describes something. Have a list of items with just an icon for example makes no sense either.

> However there are also some that actually use the key to get the descriptions for only those sensors they want to add based on the monitored_conditions. rova would be an example for that.

Monitored conditions have been deprecated a long time ago. We should not take this into account. For Rova a loop over the available sensors and only spawn matching keys will do.

Reply:
Gave a bit more thought, I'd vote to keep key as the sole exception in the base description class. If it should optional or not... I dunno, I guess there is not real harm in making it optional.

Reply:
I am a big fan of keeping it inside the class, because the alternative is that we're passing around 2 values. And in the end each description needs a unique identifier, which is what `key` is.

Reply:
Agreed. Additionally, if unsolvable use-cases appear, we can always make it optional later (the other way around is harder).